standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -156,13 +157,21 @@ private ReplChangeManager(Configuration conf) throws MetaException {
           //Create default cm root
           Path cmroot = new Path(cmRootDir);
           createCmRoot(cmroot);
+          FileSystem cmRootFs = cmroot.getFileSystem(conf);
           HdfsEncryptionShim pathEncryptionShim = hadoopShims
-                  .createHdfsEncryptionShim(cmroot.getFileSystem(conf), conf);
+                  .createHdfsEncryptionShim(cmRootFs, conf);
           if (pathEncryptionShim.isPathEncrypted(cmroot)) {
-            String encryptionZonePath = pathEncryptionShim.getEncryptionZoneForPath(cmroot).getPath();
-            encryptionZones.put(encryptionZonePath, cmroot);
+            //cm root cannot be encrypted. So we move the current cmroot data to a cmroot in a encrypted zone
+            String encryptionZonePath = cmRootFs.getUri()
+                    + pathEncryptionShim.getEncryptionZoneForPath(cmroot).getPath();
+            cmRootFs.rename(cmroot, new Path(encryptionZonePath + encryptedCmRootDir));
```

Comment:
validation of encryptedCmRootDir should be done first for absolute path and then we should do the move

Reply:
move will cause issue if some repl dumps are already there with old cm path

Reply:
Yes we can do a copy but that will take a lot of time.